File: external/HDF-EOS/hdfeos/samples/readgdflds.f
	program readgdflds

	integer       status, gdrdfld, gdrdattr, gddetach, gdclose
	integer*4     gdfid, gdid, gdopen, gdattach
	integer*4     start(2), stride(2), count(2)
	real*4        f32, veg(120,200)
	integer DFACC_RDWR
	parameter (DFACC_RDWR=3)
	integer DFNT_FLOAT32
	parameter (DFNT_FLOAT32=5)

	gdfid = gdopen("GridFile.hdf", DFACC_RDWR)

	if (gdfid .ne. -1) then

	   gdid = gdattach(gdfid, "UTMGrid")

	   if (gdid .ne. -1) then
	
	      start(1) = 0
	      start(2) = 0
	      stride(1) = 1
	      stride(2) = 1
	      count(1) = 120
	      count(2) = 200
	      status = gdrdfld(gdid, "Vegetation", 
     1    	               start, stride, count, veg)
	      
	      

	      status = gdrdattr(gdid, "float32", f32)
	      write(*,*) 'f32: ', f32
	      
	   endif
	endif

	status = gddetach(gdid)
	status = gdclose(gdfid)

	stop
	end

File: external/HDF-EOS/hdfeos/samples/ReadGDflds.c
#include "hdf.h"


/*
 * In this example we will (1) open the "GridFile" HDF file, (2) attach to
 * the "UTMGrid", and (3) read data from the "Vegetation" field.
 */


main()
{

    intn            i, j, status;
    
    int32           gdfid, GDid;

    float32         f32=1.0;
    float32         veg[200][120];


    /*
     * Open the HDF grid file, "GridFile.hdf".
     */

    gdfid = GDopen("GridFile.hdf", DFACC_RDWR);


    if (gdfid != -1)
    {

	/*
	 * Attach the "UTMGrid".
	 */

	GDid = GDattach(gdfid, "UTMGrid");

	if (GDid != -1)
	{
	  status = GDreadfield(GDid, "Vegetation", 
				NULL, NULL, NULL, veg);

          status = GDreadattr(GDid, "float32", &f32);
	  
	}
    }

    GDdetach(GDid);

    GDclose(gdfid);

    return;
}
